analyzers/src/SonarAnalyzer.CSharp/SymbolicExecution/Roslyn/ConditionEvaluatesToConstant.cs

```diff
@@ -30,6 +30,17 @@ public class ConditionEvaluatesToConstant : ConditionEvaluatesToConstantBase
 
     public override bool ShouldExecute() => true;
 
+    protected override bool IsLeftCoalesceExpression(SyntaxNode syntax) =>
+        syntax.Parent is BinaryExpressionSyntax { } binary
+        && binary.OperatorToken.IsKind(SyntaxKind.QuestionQuestionToken)
+        && binary.Left == syntax;
+
+    protected override bool IsConditionalAccessExpression(SyntaxNode syntax) =>
+        (syntax.Parent as ConditionalAccessExpressionSyntax)?.Expression == syntax;
+
+    protected override bool IsForLoopIncrementor(SyntaxNode syntax) =>
+        (syntax.Parent as ForStatementSyntax)?.Incrementors.Contains(syntax) is true;
```

Comment:
This should be in `CSharpSyntaxClassifier` - it's a class dedicated to violate the "do not touch syntax" rule

Reply:
I will drop or change some of these methods in a different PR I'm working on. I will skip this change request for now and implement it via `CSharpSyntaxClassifier` in the other PR.